### PR TITLE
update for NodeJS 15

### LINKED ISSUE #3382: Support Node v15
Supporting Node v15 should hopefully just be about:
 - Adding new node 15.0.0 to the list of `SUPPORTED_NODEJS_BUILD_VERSIONS` and `SUPPORTED_NODEJS_TESTS_VERSIONS`
 - Updating `node-gyp` cache artifact
 - Adding new TaskCluster tasks for Node v15.0.0, existing `v14.x` being:
```
$ ll taskcluster/test-nodejs_14x*.yml
-rw-rw-r-- 1 alex alex 737 sept.  9 10:22 taskcluster/test-nodejs_14x_16k-linux-amd64-opt.yml
-rw-rw-r-- 1 alex alex 652 sept.  9 10:22 taskcluster/test-nodejs_14x_16k-linux-amd64-prod_pbmodel-opt.yml
-rw-rw-r-- 1 alex alex 770 sept.  9 10:22 taskcluster/test-nodejs_14x_16k_multiarchpkg-linux-amd64-opt.yml
-rw-rw-r-- 1 alex alex 685 sept.  9 10:22 taskcluster/test-nodejs_14x_16k_multiarchpkg-linux-amd64-prod_pbmodel-opt.yml
-rw-rw-r-- 1 alex alex 786 sept.  9 10:22 taskcluster/test-nodejs_14x_16k_multiarchpkg-linux-tflite-opt.yml
-rw-rw-r-- 1 alex alex 663 sept.  9 10:22 taskcluster/test-nodejs_14x_16k_tflite-linux-amd64-prod-opt.yml
-rw-rw-r-- 1 alex alex 732 sept.  9 10:22 taskcluster/test-nodejs_14x_8k-linux-amd64-opt.yml
-rw-rw-r-- 1 alex alex 649 sept.  9 10:22 taskcluster/test-nodejs_14x_8k-linux-amd64-prod_pbmodel-opt.yml
-rw-rw-r-- 1 alex alex 765 sept.  9 10:22 taskcluster/test-nodejs_14x_8k_multiarchpkg-linux-amd64-opt.yml
-rw-rw-r-- 1 alex alex 682 sept.  9 10:22 taskcluster/test-nodejs_14x_8k_multiarchpkg-linux-amd64-prod_pbmodel-opt.yml
-rw-rw-r-- 1 alex alex 781 sept.  9 10:22 taskcluster/test-nodejs_14x_8k_multiarchpkg-linux-tflite-opt.yml
-rw-rw-r-- 1 alex alex 660 sept.  9 10:22 taskcluster/test-nodejs_14x_8k_tflite-linux-amd64-prod-opt.yml
-rw-rw-r-- 1 alex alex 691 sept.  9 10:22 taskcluster/test-nodejs_14x-armbian-arm64-opt.yml
-rw-rw-r-- 1 alex alex 558 sept.  9 10:22 taskcluster/test-nodejs_14x-darwin-amd64-opt.yml
-rw-rw-r-- 1 alex alex 724 sept.  9 10:22 taskcluster/test-nodejs_14x_multiarchpkg-armbian-arm64-opt.yml
-rw-rw-r-- 1 alex alex 590 sept.  9 10:22 taskcluster/test-nodejs_14x_multiarchpkg-darwin-amd64-opt.yml
-rw-rw-r-- 1 alex alex 606 sept.  9 10:22 taskcluster/test-nodejs_14x_multiarchpkg-darwin-tflite-opt.yml
-rw-rw-r-- 1 alex alex 715 sept.  9 10:22 taskcluster/test-nodejs_14x_multiarchpkg-raspbian-rpi3-opt.yml
-rw-rw-r-- 1 alex alex 579 sept.  9 10:22 taskcluster/test-nodejs_14x_multiarchpkg-win-amd64-opt.yml
-rw-rw-r-- 1 alex alex 586 sept.  9 10:22 taskcluster/test-nodejs_14x_multiarchpkg-win-cuda-opt.yml
-rw-rw-r-- 1 alex alex 595 sept.  9 10:22 taskcluster/test-nodejs_14x_multiarchpkg-win-tflite-opt.yml
-rw-rw-r-- 1 alex alex 681 sept.  9 10:22 taskcluster/test-nodejs_14x-raspbian-rpi3-opt.yml
-rw-rw-r-- 1 alex alex 544 sept.  9 10:22 taskcluster/test-nodejs_14x-win-amd64-opt.yml
```

### PR DESCRIPTION
**Pull request by ftyers**
This will fix #3382.

### CODE REVIEW

File: taskcluster/tc-all-vars.sh [hunk 1]
```diff
@@ -87,7 +87,7 @@ SUPPORTED_PYTHON_VERSIONS=${SUPPORTED_PYTHON_VERSIONS:-3.5.8:ucs2 3.6.10:ucs2 3.
 #
 # Also, builds should always target first major branch release to ensure proper binary compatibility
 # and tests should always target latest major branch release version
-SUPPORTED_NODEJS_BUILD_VERSIONS=${SUPPORTED_NODEJS_BUILD_VERSIONS:-10.0.0 11.0.0 12.7.0 13.0.0 14.0.0}
-SUPPORTED_NODEJS_TESTS_VERSIONS=${SUPPORTED_NODEJS_TESTS_VERSIONS:-10.20.1 11.15.0 12.17.0 13.14.0 14.3.0}
+SUPPORTED_NODEJS_BUILD_VERSIONS=${SUPPORTED_NODEJS_BUILD_VERSIONS:-10.0.0 11.0.0 12.7.0 13.0.0 14.0.0, 15.0.0}
```
Comment (lissyx) [inline taskcluster/tc-all-vars.sh:90]: no comma

File: taskcluster/tc-all-vars.sh [hunk 2]
```diff
@@ -87,7 +87,7 @@ SUPPORTED_PYTHON_VERSIONS=${SUPPORTED_PYTHON_VERSIONS:-3.5.8:ucs2 3.6.10:ucs2 3.
...
+SUPPORTED_NODEJS_TESTS_VERSIONS=${SUPPORTED_NODEJS_TESTS_VERSIONS:-10.20.1 11.15.0 12.17.0 13.14.0 14.3.0, 15.0.0}
```
Comment (lissyx) [inline taskcluster/tc-all-vars.sh:91]: no comma

File: taskcluster/.shared.yml [hunk 3]
```diff
@@ -79,8 +79,8 @@ nodejs:
     prep_14: '/usr/bin/wget.exe https://nodejs.org/dist/v14.3.0/node-v14.3.0-win-x64.zip && ""C:\Program Files\7-zip\7z.exe"" x -o$TASKCLUSTER_NODE_DIR -tzip -aoa node-v14.3.0-win-x64.zip && rm node-*.zip && export PATH=$TASKCLUSTER_TASK_DIR/bin/node-v14.3.0-win-x64/:$PATH'
```
Comment (lissyx) [inline taskcluster/.shared.yml:79]: you lack all `prep_15` entries as well

File: taskcluster/tc-all-vars.sh (hunk 1 above)
Comment (ftyers) [reply taskcluster/tc-all-vars.sh]: Fixed.

File: taskcluster/.shared.yml (hunk 3 above)
Comment (ftyers) [reply taskcluster/.shared.yml:82]: Fixed.

File: taskcluster/tc-all-vars.sh (hunk 2 above)
Comment (ftyers) [reply taskcluster/tc-all-vars.sh]: Fixed.

File: native_client/javascript/abi_crosswalk_priv.json [hunk 4]
```diff
@@ -0,0 +1,2206 @@
...
+  "14.2.0": {
+    "node_abi": 83,
+    "v8": "8.1"
+  },
+  "14.3.0": {
+    "node_abi": 83,
+    "v8": "8.1"
+  },
+  "15.0.0": {
+    "node_abi": 84,
```
Comment (lissyx) [inline native_client/javascript/abi_crosswalk_priv.json:2203]: Are you sure this should not be `88` ?

Comment (lissyx) [reply native_client/javascript/abi_crosswalk_priv.json:2203]: CI failing with `Error: Cannot find module '/home/build-user/ds-test/node_modules/deepspeech/lib/binding/v0.9.0-alpha.10/linux-x64/node-v88/deepspeech.node'` and `NODE_MODULE_VERSION` for v15 is `88` on https://nodejs.org/en/download/releases/

Comment (ftyers) [reply native_client/javascript/abi_crosswalk_priv.json]: fixed.

**Review by lissyx — APPROVED** [review]
@ftyers Please squash all of that into one commit, and then we can merge. Once it is done, please cherry-pick in a new PR for `r0.9` branch!

**Pull request merged**